### PR TITLE
Phone signup validation

### PR DESCRIPTION
**Pull request by DXCanas**
Follows the same standards as @jamalex - take anything, as long as there are 10 digits.
Also changed disabled behavior of submit button so that it's always visible per @christianmemije 
Also updated strings caps and messages per @jamalex @aronasorman 
image

### CODE REVIEW

File: kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue [hunk 1]
```diff
@@ -31,22 +31,22 @@
...
+        @blur="phoneNumberVisited=true"
+        :error="phoneNumberError"
+        :invalid="!phoneNumberValid"
+        v-model="phoneNumber"
+        autocomplete="tel"
         required
-        id="username"
-        type="text" />
+        type="tel">
+      </core-textbox>
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue]: self-closing tag.

File: kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue [hunk 2]
```diff
@@ -114,14 +113,15 @@
...
-      createAccount: 'Create an Account',
+      createAccount: 'Create an account',
       name: 'Name',
-      enterName: 'Enter Name',
-      username: 'Phone Number',
-      enterUsername: 'Enter Phone Number',
+      enterName: 'Enter name',
+      phoneNumber: 'Phone number',
+      enterPhoneNumber: 'Enter phone number',
+      phoneNumberInvalidError: 'Please enter a valid ten-digit phone number',
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue]: '10 digit' instead. https://material.io/guidelines/style/writing.html#writing-capitalization-punctuation
> Use numerals in place of words for numbers.

File: kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue [hunk 3]
```diff
@@ -114,14 +113,15 @@
...
       password: 'Password',
-      enterPassword: 'Enter Password',
-      confirmPassword: 'Confirm Password',
+      enterPassword: 'Enter password',
+      confirmPassword: 'Confirm password',
       passwordMatchError: 'Passwords do not match',
       genericError: 'Something went wrong during sign up!',
```
Comment (christianmemije) [inline kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue]: > Avoid exclamation points

**Comment by christianmemije** [comment]
`passwordMatchError` not showing up
image

**Comment by christianmemije** [comment]
No feedback given to user when attempting to submit without having clicked the checkbox for the terms of service.

**Comment by christianmemije** [comment]
The error message mentions `username` but instead should say `phone number`?
image

**Comment by christianmemije** [comment]
I seem to be able to input any string even without any digits.

**Comment by christianmemije** [comment]
It would be nice to dismiss the `ui-alert` after clicking the submit button.

**Comment by DXCanas** [comment]
@christianmemije re:no password error

Has to do with some missing props in our textbox wrapper. Going to be sending in a PR for that later today, while I wrap all existing `ui-textbox`es.

**Comment by DXCanas** [comment]
@christianmemije re:error message says username instead of phone number

That's a backend error. I think they're working on a separate API for vf? @ralphiee22

**Comment by DXCanas** [comment]
Re: dismissing alert on submit

Aren't you moving on to the next page anyway? or are you assuming that a user messes up several times/submit?

**Comment by DXCanas** [comment]
Re: submitting no digits whatsoever

I'll look into it. Does sending that bring up a backend error?

**Comment by DXCanas** [comment]
@christianmemije re: checkbox not being alerted when not populated
Seems like the required tag isn't working as expected; either HTML5 validation isn't picking it up or keen's implementation doesn't lend itself to HTML validation.

I'll look into it

**Comment by DXCanas** [comment]
@christianmemije Keen's props for `checkbox` don't include `required` or `error`. I think I'm going to either create a wrapper or submit a PR to Keen. @indirectlylit Do you have any other options or alternatives?

Either way, I think that's a little out of the scope of this PR.

**Comment by christianmemije** [comment]
> Aren't you moving on to the next page anyway? or are you assuming that a user messes up several times/submit? In this case, doesn't the error message update with each error?

Yeah assume that the user messes up multiple times with the same field.

**Comment by DXCanas** [comment]
>  In this case, doesn't the error message update with each error?

Dismissing the textbox after submission only to have it display a new error seems a bit redundant, especially if the user is making a mistake in the fields that have on-the-fly feedback

**Comment by DXCanas** [comment]
Ready for testing once [this PR in the main Kolibri repo](https://github.com/learningequality/kolibri/pull/998) is merged.

File: kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue [hunk 4]
```diff
@@ -31,22 +31,21 @@
...
-        :placeholder="$tr('enterUsername')"
-        :label="$tr('username')"
-        :aria-label="$tr('username')"
-        :invalid="usernameError"
-        v-model="username"
-        autocomplete="username"
+        :placeholder="$tr('enterPhoneNumber')"
+        :label="$tr('phoneNumber')"
+        :aria-label="$tr('phoneNumber')"
+        @blur="phoneNumberVisited=true"
```
Comment (indirectlylit) [inline kolibri_instant_schools_plugin/assets/src/vue/sign-up-page/index.vue]: note - need to add this event to `kolibri`

**Pull request merged**